pkg/deployments/tasks/20230217-gnosis-root-gauge-factory/test/task.fork.ts

```diff
@@ -0,0 +1,326 @@
+import { range } from 'lodash';
+import { expectTransferEvent } from '@balancer-labs/v2-helpers/src/test/expectTransfer';
+import { describeForkTest } from '../../../src/forkTests';
+import { deployedAt } from '@balancer-labs/v2-helpers/src/contract';
+import { WeightedPoolEncoder } from '@balancer-labs/balancer-js';
+
+describeForkTest('GnosisRootGaugeFactory', 'mainnet', 16627100, function () {
+  let veBALHolder: SignerWithAddress, admin: SignerWithAddress, recipient: SignerWithAddress;
+  let factory: Contract, gauge: Contract;
+  let vault: Contract,
+    authorizer: Contract,
+    authorizerAdaptor: Contract,
+    BALTokenAdmin: Contract,
+    gaugeController: Contract,
+    gaugeAdder: Contract,
+    veBAL: Contract,
+    bal80weth20Pool: Contract;
+  let BAL: string;
+  let task: Task;
+
+  const GOV_MULTISIG = '0x10A19e7eE7d7F8a52822f6817de8ea18204F2e4f';
+
+  const VEBAL_POOL = '0x5c6ee304399dbdb9c8ef030ab642b10820db8f56';
+  const VAULT_BOUNTY = fp(1000);
+
+  const weightCap = fp(0.001);
+
+  before('run task', async () => {
+    task = new Task('20230217-gnosis-root-gauge-factory', TaskMode.TEST, getForkedNetwork(hre));
+    await task.run({ force: true });
+    factory = await task.deployedInstance('GnosisRootGaugeFactory');
+  });
+
+  before('advance time', async () => {
+    // This causes all voting cooldowns to expire, letting the veBAL holder vote again
+    await advanceTime(DAY * 12);
+  });
+
+  before('setup accounts', async () => {
+    admin = await getSigner(0);
+    recipient = await getSigner(1);
+
+    veBALHolder = await impersonate((await getSigner(2)).address, VAULT_BOUNTY.add(fp(5))); // plus gas
+  });
+
+  before('setup contracts', async () => {
+    const vaultTask = new Task('20210418-vault', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    vault = await vaultTask.deployedInstance('Vault');
+    authorizer = await vaultTask.instanceAt('Authorizer', await vault.getAuthorizer());
+
+    const authorizerAdaptorTask = new Task('20220325-authorizer-adaptor', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    authorizerAdaptor = await authorizerAdaptorTask.deployedInstance('AuthorizerAdaptor');
+
+    const gaugeAdderTask = new Task('20230109-gauge-adder-v3', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    gaugeAdder = await gaugeAdderTask.deployedInstance('GaugeAdder');
+
+    const balancerTokenAdminTask = new Task('20220325-balancer-token-admin', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    BALTokenAdmin = await balancerTokenAdminTask.deployedInstance('BalancerTokenAdmin');
+
+    BAL = await BALTokenAdmin.getBalancerToken();
+
+    const gaugeControllerTask = new Task('20220325-gauge-controller', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    gaugeController = await gaugeControllerTask.deployedInstance('GaugeController');
+  });
+
+  before('update balancer token admin rate', async () => {
+    await advanceTime(WEEK * 5);
+    await BALTokenAdmin.update_mining_parameters();
+  });
```

Comment:
This is also a bit odd, why move forward into the future so much? And why manually update the minim params? It'll be done automatically

Reply:
It's to make it not cross an epoch, which throws off the expected emissions and causes the "multiple week" minting test to fail. If I don't do this, it's off by 6%.